static/main.js

```diff
@@ -482,6 +482,22 @@ function earlyGetDefaultLangSetting() {
     return Settings.getStoredSettings().defaultLanguage;
 }
 
+function getDefaultLangId(subLangId, options) {
+    var defaultLangId = subLangId;
+    if (!defaultLangId) {
+        var defaultLangSetting = earlyGetDefaultLangSetting();
+        if (defaultLangSetting && options.languages[defaultLangSetting] !== undefined) {
```

Comment:
The only change here is to check that the default language is available. The rest is just code being moved to a function